### PR TITLE
List IoT Hubs in Azure viewlet

### PR DESCRIPTION
**Pull request by formulahendry**
- List Subscription
- List IoT Hubs
- Refresh 
- Scope this in Explorer, and then focus on devices tree view
- Add deployment ID in output channel when deploying at scale for IoT Edge

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -619,6 +656,14 @@
           "when": "view == iotHubDevices && viewItem == interfaces-label",
           "group": "inline"
         },
+        {
+          "command": "azure-iot-hub.setIoTHub",
```
Comment (adashen) [inline package.json]: add never for this command in command palette??

File: package.nls.json [hunk 2]
```diff
@@ -52,6 +52,10 @@
   "azure-iot-toolkit.configuration.terminalRoot.description": "(Windows Only) Replace the Windows style drive letter in the command with a Unix style root when using a custom shell as the terminal, like Bash on Windows, Git Bash or Cgywin. Example: For Bash on Windows, setting this to '/mnt/' will replace 'C:\\foo\\bar' with '/mnt/c/foo/bar'",
   "azure-iot-toolkit.configuration.treeViewAutoRefreshEnable.description": "Whether to enable auto refresh of tree view.",
   "azure-iot-toolkit.configuration.treeViewAutoRefreshIntervalInSeconds.description": "Time interval in seconds for tree view auto refresh, auto refresh has to be enabled for it to work.",
+  "azure-iot-hub.category": "Azure IoT Hub",
```
Comment (adashen) [inline package.nls.json]: also add to Japanese?

File: package.json [hunk 3]
```diff
@@ -34,6 +34,7 @@
     "*",
     "onView:iotHubDevices",
```
Comment (adashen) [inline package.json]: Please also fix the new vulnerable dependency issue

File: src/Nodes/IoTHub/IoTHubAccountTreeItem.ts [hunk 4]
```diff
@@ -0,0 +1,13 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license.
+
+import { AzureAccountTreeItemBase, ISubscriptionContext, SubscriptionTreeItemBase } from "vscode-azureextensionui";
+import { IoTHubSubscriptionTreeItem } from "./IoTHubSubscriptionTreeItem";
+
+// The root of DPS treeview, represents an Azure account
```
Comment (adashen) [inline src/Nodes/IoTHub/IoTHubAccountTreeItem.ts]: Clean the comment. It is for DPS?

File: src/Nodes/IoTHub/IoTHubResourceTreeItem.ts [hunk 5]
```diff
@@ -0,0 +1,37 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license.
+
+import { IotHubModels } from "azure-arm-iothub";
+import { AzureParentTreeItem, AzureTreeItem, TreeItemIconPath } from "vscode-azureextensionui";
+import { TreeUtils } from "../../Utility/treeUtils";
+
+// Represents a DPS resource
```
Comment (adashen) [inline src/Nodes/IoTHub/IoTHubResourceTreeItem.ts]: Clean the comment

File: src/Nodes/IoTHub/IoTHubResourceTreeItem.ts [hunk 6]
```diff
@@ -0,0 +1,37 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license.
+
+import { IotHubModels } from "azure-arm-iothub";
+import { AzureParentTreeItem, AzureTreeItem, TreeItemIconPath } from "vscode-azureextensionui";
+import { TreeUtils } from "../../Utility/treeUtils";
+
+// Represents a DPS resource
+export class IoTHubResourceTreeItem extends AzureTreeItem {
+    public static contextValue: string = "IotHub";
```
Comment (adashen) [inline src/Nodes/IoTHub/IoTHubResourceTreeItem.ts]: is it public or private?

File: src/Nodes/IoTHub/IoTHubSubscriptionTreeItem.ts [hunk 7]
```diff
@@ -0,0 +1,29 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license.
+
+// import { IotDpsClient, IotDpsModels } from "azure-arm-deviceprovisioningservices";
```
Comment (adashen) [inline src/Nodes/IoTHub/IoTHubSubscriptionTreeItem.ts]: clean comment

File: src/Nodes/IoTHub/IoTHubAccountTreeItem.ts [hunk 8]
```diff
@@ -0,0 +1,13 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
```
Comment (adashen) [inline src/Nodes/IoTHub/IoTHubAccountTreeItem.ts]: It's good to have a folder dedicated for IoTHub tree items. Please also help to move the same set of DPS treeitems to a new DPS folder?

File: src/extension.ts [hunk 9]
```diff
@@ -229,21 +227,28 @@ export function activate(context: vscode.ExtensionContext) {
 export function deactivate() {
 }
 
-function activateDps(context: vscode.ExtensionContext) {
-    let telemetryReporter = new TelemetryClientWrapper("DPS.");
-    let outputChannel = createAzExtOutputChannel("Device Provisioning Service", "azure-iot-toolkit");
+function initializeTreeView(context: vscode.ExtensionContext) {
+    const telemetryReporter = new TelemetryClientWrapper("IoTHub.");
```
Comment (adashen) [inline src/extension.ts]: So now DPS and IoTHub tree  share the same telemetryReporter with key "IoTHub"?

File: src/extension.ts [hunk 10]
```diff
@@ -229,21 +227,28 @@ export function activate(context: vscode.ExtensionContext) {
...
+    const outputChannel = createAzExtOutputChannel("Azure IoT Hub Toolkit", "azure-iot-toolkit");
```
Comment (adashen) [inline src/extension.ts]: same as above the dps and iot hub tree view use the same output channel?

File: src/iotHubResourceExplorer.ts [hunk 11]
```diff
@@ -146,15 +151,38 @@ export class IoTHubResourceExplorer extends BaseExplorer {
...
+            await Utility.storeIoTHubInfo(subscriptionItem.subscription.subscriptionId, iotHubItem.iotHubDescription);
             return iotHubItem.iotHubDescription;
         }
     }
 
+    public async setIoTHub(context: IActionContext, node?: IoTHubResourceTreeItem): Promise<void> {
+        if (!node) {
+            node = await this.iotHubTreeDataProvider.showTreeItemPicker<IoTHubResourceTreeItem>("IotHub", context);
+        }
+        this._outputChannel.show();
```
Comment (adashen) [inline src/iotHubResourceExplorer.ts]: the logic here is duplicate with the selectIoTHub, how about abstract a method to share between the commands?

File: package.json (hunk 1 above)
Comment (formulahendry) [reply package.json]: Added

File: package.nls.json (hunk 2 above)
Comment (formulahendry) [reply package.nls.json]: 'category' is not part of l18n.

File: src/Nodes/IoTHub/IoTHubAccountTreeItem.ts (hunk 4 above)
Comment (formulahendry) [reply src/Nodes/IoTHub/IoTHubAccountTreeItem.ts]: Updated

File: src/Nodes/IoTHub/IoTHubResourceTreeItem.ts (hunk 5 above)
Comment (formulahendry) [reply src/Nodes/IoTHub/IoTHubResourceTreeItem.ts]: updated

File: src/Nodes/IoTHub/IoTHubSubscriptionTreeItem.ts (hunk 7 above)
Comment (formulahendry) [reply src/Nodes/IoTHub/IoTHubSubscriptionTreeItem.ts]: cleaned

File: src/Nodes/IoTHub/IoTHubResourceTreeItem.ts (hunk 6 above)
Comment (formulahendry) [reply src/Nodes/IoTHub/IoTHubResourceTreeItem.ts]: updated to private

File: src/extension.ts (hunk 9 above)
Comment (formulahendry) [reply src/extension.ts]: Yes~

File: src/extension.ts (hunk 10 above)
Comment (formulahendry) [reply src/extension.ts]: Yes~

File: src/iotHubResourceExplorer.ts (hunk 11 above)
Comment (formulahendry) [reply src/iotHubResourceExplorer.ts]: updated

File: src/Nodes/IoTHub/IoTHubAccountTreeItem.ts (hunk 8 above)
Comment (formulahendry) [reply src/Nodes/IoTHub/IoTHubAccountTreeItem.ts]: Moved

**Pull request merged**